wallet_ui/components/panels/CreateCanister.tsx

```diff
@@ -80,15 +86,43 @@ export function CreateCanisterDialog(props: {
   function handleClose() {
     close();
   }
+  function increaseInput() {
+    setCount(count + 1);
+    setMoreControllers(prev => [...prev, ""]);
+  }
+
+  function handleInputChange(ind: number, ev: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) {
+    const newInput = moreControllers.map((controller : string, i : number) => {
+      if(ind === i) {
+        return ev.target.value;
+      }
+      return controller;
+    })
+    setMoreControllers(newInput);
+    try {
+      Principal.fromText(ev.target.value);
+      setError2(false);
```

Comment:
What prompted the need for multiple error states?

Reply:
It was due to the first controller and additional controllers being different elements. However, I'm consolidating into one, so there will be just 1 error state.